### ISSUE TITLE
[PSym] Receive inside a loop

### ISSUE THREAD

**Issue by aman-goel** [p-org/P#501]
Hi,

P compiler for symbolic backend currently does not correctly compile the symbolic IR when `receive` statement is present inside a loop.

For example,

```
event ping0: machine;
event ping1: machine;
event pong0;
event pong1;

machine Main {
  var Follower0: machine;
  var Follower1: machine;

  start state Init {
    entry {
      var id: int;
      Follower0 = new Follower();
      Follower1 = new Follower();
      id = 0;
      while(id < 3) {
        send Follower1, ping0, this;
        print format ("{0}: ping0 sent", id);
        receive {
            case pong0: {
                print format ("{0}: pong0 received", id);
           }
        }
        print format ("{0}: done", id);
        id = id + 1;
      }
    }
  }
}

machine Follower {
  start state Init {
    on ping0 do (sender: machine) {
      send sender, pong0;
    }
    on ping1 do (sender: machine) {
      send sender, pong1;
    }
  }
}
```

In the above example, PSym currently does not handle all possible cases correctly, and exits after the first loop iteration (instead of iterating through the loop 3 times).

Running this example on the fp branch (0277c74003d7ad5a2ab7891e2f8fb7ed86e972f4) produces the following output:

```
Starting Iteration: 0
[{0}: ping0 sent]
Step 0: scheduled event[createMachine] sent to [Follower(0)]
Step 1: scheduled event[createMachine] sent to [Follower(1)]
Step 2: scheduled event[ping0] sent to [Follower(1)]
Step 3: scheduled event[pong0] sent to [Main(0)]
[{0}: pong0 received]
[{0}: done]
Execution finished in 4 steps
Finished Iteration: 0: Max Depth: 0TotalTransitions = 0, ReducedTransitionsExplored = 0
Execution finished in 4 steps
Took 0 seconds
```

Commit 67b711ed8b7e68be8c1bcfa949d8c84a6ecaf80d disables receive inside a loop for now.

@ankushdesai, @lmpick Adding this as a TODO / feature request.

--
@aman-goel

**Comment by ankushdesai** [comment]
@aman-goel hey is this fixed now?

**Comment by aman-goel** [comment]
Yes, resolved with #507
